meta.go

```diff
@@ -42,11 +42,19 @@ type ServiceMeta struct {
 	Stdout        ServiceLog       `json:"stdout"`
 	Stderr        ServiceLog       `json:"stderr"`
 	Type          ServiceType
+	ExitCode      int  `json:"exit_code"`
 	Ignore        bool `json:"ignore"`
 	IgnoreFailure bool `json:"ignore_failure"`
 	Selected      bool
 }
 
+func (m *ServiceMeta) GetExitCode() int {
+	if m.ExitCode < 1 {
```

Comment:
Shouldn't this only test for `m.ExitCode == 0` because that's the only non error code and a negative exit code would also indicate an error? Also why does it return 1 for the 0 case? I find this quite confusing

Reply:
For the `== 0` part: As an integer defaults to 0 if not set, without this line all steps would indicate success.
For the `< 0` part: I never found a valid definition of an exit code `< 0`, e.g. `-1` was only used for information internally to signify that the exit code can not be trusted. The golang doc for [`os.Exit`](https://golang.org/pkg/os/#Exit) also states:
> For portability, the status code should be in the range [0, 125].

Reply:
Thinking a bit more about it, I think I can remove the custom type and rework this part a bit.